code/modules/mob/living/carbon/xenomorph/damage_procs.dm

```diff
@@ -3,21 +3,23 @@
 		return
 	return ..()
 
+/mob/living/carbon/xenomorph/modify_by_armor(damage_amount, armor_type, penetration, def_zone)
+	var/hard_armor_modifier = get_hard_armor(armor_type, def_zone)
+	hard_armor_modifier = hard_armor_modifier - (hard_armor_modifier * penetration * 0.01)
+	var/soft_armor_modifier = min((1 - ((get_soft_armor(armor_type, def_zone) - penetration) * 0.01)), 1)
+	return clamp(((damage_amount - hard_armor_modifier) * soft_armor_modifier), 0, damage_amount)
+
 /mob/living/carbon/xenomorph/ex_act(severity)
 	if(status_flags & (INCORPOREAL|GODMODE))
 		return
 
-
 	var/bomb_armor = soft_armor.getRating("bomb")
-	var/bomb_effective_armor = (bomb_armor/100)*get_sunder()
-	var/bomb_damage_multiplier = max(0, 1 - bomb_effective_armor)
-	var/bomb_slow_multiplier = max(0, 1 - 3.5*bomb_effective_armor)
-	var/bomb_sunder_multiplier = max(0, 1 - bomb_effective_armor)
-
-
 	if(bomb_armor >= 100)
 		return //immune
 
+	var/bomb_slow_multiplier = modify_by_armor(0.7, BOMB)
```

Comment:
Some balance effect here, or am I mistaken ?

Reply:
whoops. I was going to redo status effect stuff in the same pr before deciding to atomise, must have slipped through. P.S. status effect interactions with armor are very weird and arbitrary right now.